File: cdk_integration_tests/src/python/S3BucketKMSEncryption/fail__3__.py
from aws_cdk import App, Stack, aws_s3 as s3


class AppStack(Stack):
    def __init__(self, app: App, id: str) -> None:
        super().__init__(app, id)

        fail_1 = s3.Bucket(
            self,
            "example",
        )

        fail_2 = s3.Bucket(
            self,
            "example",
            encryption=s3.BucketEncryption.UNENCRYPTED,
        )

        fail_3 = s3.Bucket(
            self,
            "example",
            encryption=s3.BucketEncryption.S3_MANAGED,
        )


app = App()
AppStack(app, "example-stack")
app.synth()

File: cdk_integration_tests/src/typescript/S3BucketKMSEncryption/fail__3__.ts
import * as cdk from 'aws-cdk-lib';
import * as s3 from 'aws-cdk-lib/aws-s3';
import { Construct } from 'constructs';

export class exampleStack extends cdk.Stack {
  constructor(scope: Construct, id: string, props?: cdk.StackProps) {
    super(scope, id, props);

    const fail = new s3.Bucket(this, 'example', {});

    const fail2 = new s3.Bucket(this, 'example', {
      encryption: s3.BucketEncryption.UNENCRYPTED
    });

    const fail3 = new s3.Bucket(this, 'example', {
      encryption: s3.BucketEncryption.S3_MANAGED
    });
  }
}

const app = new cdk.App();
new exampleStack(app, 'example-stack');
app.synth();
